src/dune_lang/atom.ml

```diff
@@ -45,6 +45,12 @@ let of_string s = A s
 
 let to_string (A s) = s
 
+let of_valid_string s =
```

Comment:
The name `of_valid_string` suggests to me that the string is assumed to be valid (rather than checked).

Maybe this could be called `parse`?

Reply:
Fixed, thanks!